packages/frontend/editor-ui/src/components/NDVSubConnections.vue

```diff
@@ -203,7 +207,16 @@ function onPlusClick(connectionKey: string) {
 		return;
 	}
 
-	emit('openConnectionNodeCreator', props.rootNode.name, type as NodeConnectionType);
+	// Extract the index from connectionKey
+	const [, indexStr] = connectionKey.split('-');
+	const connectionIndex = parseInt(indexStr, 10) || 0;
```

Comment:
This could be better if we passed an object with a clear interface instead of a string that's parsed here instead. 
For example
```
type ConnectionPosition = {type: NodeConnectionType; index: number};
getConnectionInfo(connection: INodeInputConfiguration, globalIndex: number): ConnectionPosition; // returns an object with clear interface, could be used with onPlusClick handler for example
getConnectionKey(connectionInfo: ConnectionPosition): string; // can be used in other places where a string is needed, like `:key`
```
This would make it clearer and increase typesafety.
`type as NodeConnectionType,` would become unnecessary, which is more typesafe.
`parseInt(indexStr, 10) || 0;` more typesafe because we don't have risk of `NaN`.